spec/unit/check_game_state_spec.rb

```diff
@@ -11,46 +11,46 @@
 
   it 'can show when player one wins from a row' do
     game_state_gateway.game_state = [1, 1, 1, 0, 0, 0, 0, 0, 0]
-    expect(check_game_state.execute).to eq('Player one wins')
+    expect(check_game_state.execute).to eq(1)
   end
 
   it 'can show when player two wins from a row' do
     game_state_gateway.game_state = [2, 2, 2, 0, 0, 0, 0, 0, 0]
-    expect(check_game_state.execute).to eq('Player two wins')
+    expect(check_game_state.execute).to eq(2)
   end
 
   it 'can show winner when player one wins from a different row' do
     game_state_gateway.game_state = [0, 0, 0, 1, 1, 1, 0, 0, 0]
-    expect(check_game_state.execute).to eq('Player one wins')
+    expect(check_game_state.execute).to eq(1)
   end
 
   it 'can show when player one wins from a column' do
     game_state_gateway.game_state = [1,0,0,1,0,0,1,0,0]
-    expect(check_game_state.execute).to eq('Player one wins')
+    expect(check_game_state.execute).to eq(1)
   end
 
   it 'can show when player two wins from a column' do
     game_state_gateway.game_state = [2,0,0,2,0,0,2,0,0]
-    expect(check_game_state.execute).to eq('Player two wins')
+    expect(check_game_state.execute).to eq(2)
   end
 
   it 'can show winner when player one wins from a different column' do
     game_state_gateway.game_state = [0, 1, 0, 0, 1, 0, 0, 1, 0]
-    expect(check_game_state.execute).to eq('Player one wins')
+    expect(check_game_state.execute).to eq(1)
   end
 
   it 'can show winner when player one wins from a diagonal' do
     game_state_gateway.game_state = [1, 0, 0, 0, 1, 0, 0, 0, 1]
-    expect(check_game_state.execute).to eq('Player one wins')
+    expect(check_game_state.execute).to eq(1)
   end
 
   it 'can show winner when player two wins from a diagonal' do
     game_state_gateway.game_state = [2, 0, 0, 0, 2, 0, 0, 0, 2]
-    expect(check_game_state.execute).to eq('Player two wins')
+    expect(check_game_state.execute).to eq(2)
   end
 
   it 'can show winner when player one wins from a different diagonal' do
     game_state_gateway.game_state = [0, 0, 1, 0, 1, 0, 1, 0, 0]
-    expect(check_game_state.execute).to eq('Player one wins')
+    expect(check_game_state.execute).to eq(1)
```

Comment:
What value do you return if there is `a tie`? This is a zero-sum game.

Reply:
We will be implementing draws after merging this branch. The idea is to check that the array doesn't contain a 0 at any index and the array is filled with 1 and 2 values - at all 9 indexes. We can return a value in this case e.g. 0 or 3 depending on which is better